### PR TITLE
[SYCL][ESIMD]Implement gather(lacc) accepting compile time properties

### PR DESCRIPTION
**Pull request by fineg74**
This implements the unified memory API for gather with local accessors

### CODE REVIEW

File: sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp:815]: nit: if use `SomeVar` style, then
```suggestion
             simd<uint32_t, 8> Offsets(I * sizeof(T), sizeof(T));
```

**Review by v-klochkov — COMMENTED** [review]
Looks really good! I cannot add anything except fixing 1 letter ))
@sarnex - please also take a look, I could miss something.

Assume approved, but pending to review/approval from Nick. Thank you.

File: sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp:815]: +1 more comment: Please ensure there are calls and compile-tests like: 
gather(lacc, byte_offsets, 0), and gather(lacc, byte_offsets, 0, mask)
to verify lacc-ga-4,5  having MaskT template parameter is sufficient work-around for 2 older functions having 'glob_offsets' parameter.

**Review by sarnex — APPROVED** [review]
Really sorry, I promised I'd do this yesterday but something came up.

LGTM with Slava's comments

**Review by sarnex — APPROVED** [review]

File: sycl/test/esimd/memory_properties.cpp
Comment (v-klochkov) [inline sycl/test/esimd/memory_properties.cpp:1222]: nit:
```suggestion
  // Validate that a new API doesn't conflict with the old API.
```

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**
